### PR TITLE
Extend rcll-refbox-instruct to wait for a given number of robots

### PR DESCRIPTION
**Pull request by morxa**
Sometimes, we may want to wait for robots, e.g., when running in
simulation. Instead of guessing how long the robots need to connect,
extend the rcll-refbox-instruct tool to wait for some given number of
robots. This also works in conjunction with the other wait options,
e.g., waiting for a game phase.

Example usage:
```
$ ./rcll-refbox-instruct -n2 -W -p PRODUCTION
```
This will wait for two robots to appear, as well as for the phase to switch to PRODUCTION.

### CODE REVIEW

File: src/tools/rcll-refbox-instruct.cpp [hunk 1]
```diff
@@ -139,18 +145,29 @@ handle_message(uint16_t                                   component_id,
 	}
 
 	std::shared_ptr<GameState> g;
-	if ((g = std::dynamic_pointer_cast<GameState>(msg)) && wait_state_) {
+	if ((g = std::dynamic_pointer_cast<GameState>(msg))) {
+		current_game_state_ = g;
+	}
+	std::shared_ptr<RobotInfo> ri;
+	if ((ri = std::dynamic_pointer_cast<RobotInfo>(msg)) && wait_state_) {
```
Comment (TarikViehmann) [inline src/tools/rcll-refbox-instruct.cpp]: the condition about `wait_state_` is not needed here

Comment (morxa) [reply src/tools/rcll-refbox-instruct.cpp]: Good point.

**Comment by morxa** [comment]
@TarikViehmann Thanks for pointing to the unnecessary check, can you re-approve? Thanks!

**Comment by morxa** [comment]
Thanks for reviewing!

**Pull request merged**